infra/runtests.py

```diff
@@ -92,6 +101,15 @@ def run_tests(chapter, file_name):
                 print(f"  {t.bold(value)}: Testing {chapter}...", end=" ")
                 results[value] = run_tests(chapter, value)
                 if not results[value][0]: print(t.green("pass"))
+            elif key == "full":
+                print(f"  {t.bold(value)}: Testing inlined {chapter}...", end=" ")
+                full_file = value.replace("-tests.md", ".full.py")
+                mod_name = value.replace("-tests.md", "")
+                os.system("make -C .. src/" + full_file)
```

Comment:
This is kinda ugly, an alternative would be to add a make dependency

Reply:
How about we change all but lab10-tests and lab16-tests to just import the full inlined code version directly rather than the non-inlined one?

Reply:
Eh. Then we're still trusting the inliner in some way.

Reply:
The ugliness here, to be clear, is that the tester calls Make to build the inlined version. Not that I think about it, it's probably possible to call the inliner through Python & avoid any impacts on the file system.

Reply:
Ok, I messed around a bit with the internals of `import` in Python so now this ugliness is gone. `runtests.py` just uses `asttools` to produce an inlined AST and then imports that AST without having to write anything to disk.